ansible/inventory/multi_inventory.py

```diff
@@ -5,29 +5,294 @@
+        if self.config.has_key('cache_max_age'):
+            self.cache_max_age = self.config['cache_max_age']
+
+        # set to global account cache_path
+        self.cache_path = cache_path
+        # if an account setting exists, set to account cache_path
+        if self.config.has_key('cache_path'):
+            self.cache_path = self.config['cache_path']
+
+        # collect a list of clusters for this account
+        self.clusters = []
+        if self.config.has_key('cluster_vars') and \
+            self.config['cluster_vars'].has_key('clusters'):
+            self.clusters = self.config['cluster_vars']['clusters'].keys()
+
+    @property
+    def inventory(self):
+        """property for inventory"""
+        if self._inventory == None:
+            self._inventory = MultiInventoryUtils.get_inventory_from_cache(os.path.join(self.cache_path, self.name))
+            self.apply_account_config()
+
+        return self._inventory
+
+    @inventory.setter
+    def inventory(self, data):
+        """setter for inventory"""
+        self._inventory = data
+        self.apply_account_config()
+
+    @staticmethod
+    def run_account_providers(accounts, force=False, validate_cache=True, account=None, debug=False):
+        """Setup the provider call with proper variables and call the account's run_provider"""
+        try:
+            updated = False
+            # if account parameter is present we only want refresh that specific account
+            tmp_accounts = accounts
+            if account:
+                tmp_accounts = [acc for acc in accounts if acc.name == account]
+
+            processes = {}
+            for acc in tmp_accounts:
+                if debug:
+                    print "Calling run_provider for account: %s" % acc.name
+                processes[acc.name] = acc.run_provider(force=force, validate_cache=validate_cache)
+
+            # for each process collect stdout when its available
+            for acc in tmp_accounts:
+                if not processes[acc.name]['cached']:
+                    proc = processes[acc.name]['proc']
+                    out, err = proc.communicate()
+
+                    # For any non-zero, raise an error on it
+                    if proc.returncode != 0:
+                        err_msg = ['\nProblem fetching account: %s' % acc.name,
+                                   'Error Code: %s' % proc.returncode,
+                                   'StdErr: %s' % err,
+                                   'Stdout: %s' % out,
+                                  ]
+                        raise RuntimeError('\n'.join(err_msg))
+
+                    updated = True
+                    # The reason this exists here is that we
+                    # are using the futures pattern.  The account object
+                    # has run a subprocess and we are collecting the results.
+                    acc.inventory = json.loads(out)
+                    acc.write_to_cache()
```

Comment:
Consider using a temp var to capture json.loads and pass to write_to_cache so we get a pre-config applied cache.